packages/block-editor/src/components/list-view/index.js

```diff
@@ -1,281 +1,27 @@
-	const [ fixedListWindow ] = useFixedWindowList(
-		elementRef,
-		BLOCK_LIST_ITEM_HEIGHT,
-		visibleBlockCount,
-		{
-			useWindowing: true,
-			windowOverscan: 40,
-		}
-	);
-
-	const expand = useCallback(
-		( clientId ) => {
-			if ( ! clientId ) {
-				return;
-			}
-			setExpandedState( { type: 'expand', clientIds: [ clientId ] } );
-		},
-		[ setExpandedState ]
-	);
-	const collapse = useCallback(
-		( clientId ) => {
-			if ( ! clientId ) {
-				return;
-			}
-			setExpandedState( { type: 'collapse', clientIds: [ clientId ] } );
-		},
-		[ setExpandedState ]
-	);
-	const expandRow = useCallback(
-		( row ) => {
-			expand( row?.dataset?.block );
-		},
-		[ expand ]
-	);
-	const collapseRow = useCallback(
-		( row ) => {
-			collapse( row?.dataset?.block );
-		},
-		[ collapse ]
-	);
-	const focusRow = useCallback(
-		( event, startRow, endRow ) => {
-			if ( event.shiftKey ) {
-				updateBlockSelection(
-					event,
-					startRow?.dataset?.block,
-					endRow?.dataset?.block
-				);
-			}
-		},
-		[ updateBlockSelection ]
-	);
-
-	const contextValue = useMemo(
-		() => ( {
-			isTreeGridMounted: isMounted.current,
-			draggedClientIds,
-			expandedState,
-			expand,
-			collapse,
-			BlockSettingsMenu,
-			listViewInstanceId: instanceId,
-			renderAdditionalBlockUI,
-		} ),
-		[
-			draggedClientIds,
-			expandedState,
-			expand,
-			collapse,
-			BlockSettingsMenu,
-			instanceId,
-			renderAdditionalBlockUI,
-		]
-	);
-
-	// If there are no blocks to show, do not render the list view.
-	if ( ! clientIdsTree.length ) {
-		return null;
-	}
-
-	return (
-		<AsyncModeProvider value={ true }>
-			<ListViewDropIndicator
-				listViewRef={ elementRef }
-				blockDropTarget={ blockDropTarget }
-			/>
-			<TreeGrid
-				id={ id }
-				className="block-editor-list-view-tree"
-				aria-label={ __( 'Block navigation structure' ) }
-				ref={ treeGridRef }
-				onCollapseRow={ collapseRow }
-				onExpandRow={ expandRow }
-				onFocusRow={ focusRow }
-				applicationAriaLabel={ __( 'Block navigation structure' ) }
-				// eslint-disable-next-line jsx-a11y/aria-props
-				aria-description={ description }
-			>
-				<ListViewContext.Provider value={ contextValue }>
-					<ListViewBranch
-						blocks={ clientIdsTree }
-						parentId={ rootClientId }
-						selectBlock={ selectEditorBlock }
-						showBlockMovers={ showBlockMovers }
-						fixedListWindow={ fixedListWindow }
-						selectedClientIds={ selectedClientIds }
-						isExpanded={ isExpanded }
-						shouldShowInnerBlocks={ shouldShowInnerBlocks }
-						showAppender={ showAppender }
-					/>
-				</ListViewContext.Provider>
-			</TreeGrid>
-		</AsyncModeProvider>
-	);
-}
-export const PrivateListView = forwardRef( ListViewComponent );
+const { PrivateListView } = unlock( blockEditorPrivateApis );
 
 export default forwardRef( ( props, ref ) => {
 	return (
 		<PrivateListView
 			ref={ ref }
 			{ ...props }
 			showAppender={ false }
-			blockSettingsMenu={ BlockSettingsDropdown }
```

Comment:
I removed this in my commit because `BlockSettingsDropdown` is the default value

Reply:
Yeah we can't do that because the point it to restrict this from being prop being used by third parties. `undefined` should work though, I'll push that now.

Reply:
Done in 74a8ac27037f87b7daf001213565aee627e1e731